contracts/interchain-token-service/src/contract.rs

```diff
@@ -281,45 +288,32 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         destination_chain: String,
         gas_token: Token,
     ) -> Result<BytesN<32>, ContractError> {
-        caller.require_auth();
-
         let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
-        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
-        let token_address = env
-            .storage()
-            .persistent()
-            .get::<_, TokenIdConfigValue>(&DataKey::TokenIdConfigKey(token_id.clone()))
-            .ok_or(ContractError::InvalidTokenId)?
-            .token_address;
-        let token = token::Client::new(env, &token_address);
-        let token_metadata = TokenMetadata {
-            name: token.name(),
-            decimal: token.decimals(),
-            symbol: token.symbol(),
-        };
-
-        assert_ok!(validate_token_metadata(token_metadata.clone()));
-
-        let message = Message::DeployInterchainToken(DeployInterchainToken {
-            token_id: token_id.clone(),
-            name: token_metadata.name.clone(),
-            symbol: token_metadata.symbol.clone(),
-            decimals: token_metadata.decimal as u8,
-            minter: None,
-        });
+        Self::_deploy_remote_interchain_token(
+            env,
+            caller,
+            deploy_salt,
+            destination_chain,
+            gas_token,
+        )
+    }
 
-        InterchainTokenDeploymentStartedEvent {
-            token_id: token_id.clone(),
-            token_address,
-            destination_chain: destination_chain.clone(),
-            name: token_metadata.name,
-            symbol: token_metadata.symbol,
-            decimals: token_metadata.decimal,
-            minter: None,
-        }
-        .emit(env);
+    fn deploy_remote_canonical_token(
+        env: &Env,
+        caller: Address,
```

Comment:
caller is not relevant here. see EVM reference. Remove caller from underlying helper function